File: java/src/main/java/io/hron/ast/TimeOfDay.java
package io.hron.ast;

/**
 * Represents a time of day (hour and minute).
 *
 * @param hour the hour (0-23)
 * @param minute the minute (0-59)
 */
public record TimeOfDay(int hour, int minute) {
  /**
   * Returns the time as total minutes from midnight.
   *
   * @return total minutes from midnight
   */
  public int totalMinutes() {
    return hour * 60 + minute;
  }

  @Override
  public String toString() {
    return String.format("%02d:%02d", hour, minute);
  }
}

File: csharp/Hron/Ast/TimeOfDay.cs
namespace Hron.Ast;

/// <summary>
/// Represents a time of day (hour and minute).
/// </summary>
/// <param name="Hour">The hour (0-23)</param>
/// <param name="Minute">The minute (0-59)</param>
public readonly record struct TimeOfDay(int Hour, int Minute)
{
    /// <summary>
    /// Returns the time as total minutes from midnight.
    /// </summary>
    public int TotalMinutes => Hour * 60 + Minute;

    public override string ToString() => $"{Hour:D2}:{Minute:D2}";
}
